test/sources/npm_test.rb

```diff
@@ -80,5 +80,27 @@
         end
       end
     end
+
+    describe "missing dependencies (glob is missing package)" do
+      let(:fixtures) { File.expand_path("../../fixtures/npm/missing-dependencies", __FILE__) }
+
+      it "includes missing dependencies when yarn.lock is present" do
+        Dir.chdir fixtures do
+          File.unlink("yarn.lock") if File.exist? "yarn.lock"
+
+          assert source.dependencies.detect { |dep| dep.name == "autoprefixer" }
+          assert source.dependencies.detect { |dep| dep.name == "glob" }
+        end
+      end
+
+      it "excludes missing dependencies when yarn.lock is present" do
+        Dir.chdir fixtures do
+          File.write("yarn.lock", "") unless File.exist? "yarn.lock"
```

Comment:
This yarn lockfile needs to be deleted as part of this test's cleanup.  It's 👍 that the other added test accounts for it, but none of the other tests in this file do and likely any other tests added in the future won't either.  It could cause unexpected behavior if the file sticks around past the end of the test 🙇

Reply:
N/A after test rework.